datashare-tasks/src/main/java/org/icij/datashare/asynctasks/TaskManager.java

```diff
@@ -58,6 +65,35 @@ default List<Task<?>> getTasks(User user, Map<String, Pattern> filters, WebQuery
         taskStream = getFilteredTaskStream(filters, taskStream);
         return taskStream.filter(t -> user.equals(t.getUser())).skip(pagination.from).limit(pagination.size).collect(toList());
     }
+
+    default List<Task<?>> getTasks(User user, Map<String, Pattern> filters, WebQueryPagination pagination, List<BatchSearchRecord> batchSearchRecords) throws IOException {
```

Comment:
I'm wondering if the addition of this API would be better located in `datashare-app` since it's really an applicative API/patch and not a generic `datashare-tasks` API.

You could probably add a `DatashareTaskManager` interface in `datashare-app` extending `TaskManager` and then use this class in `datashare-app` instead of the `TaskManager`

Reply:
Thanks for the suggestion! As agreed, this should be done in another refactoring PR.